batch/test/test_batch.py

```diff
@@ -420,3 +422,35 @@ def test_port(self):
         batch = b.wait()
         print(j.log())
         assert batch['state'] == 'success', batch
+
+    def test_restartable_insert(self):
+        builder = self.client.create_batch()
+        i = 0
+
+        def every_third_time():
+            nonlocal i
+            i += 1
+            if i % 3 == 0:
+                return True
+            return False
+        for i in range(9):
+            builder.create_job('ubuntu:18.04', ['echo', 'a'])
+
+        real_session = builder._async_builder._client._session
+        builder._async_builder._client._session = FailureInjectingClientSession(
+            builder._async_builder._client._session, every_third_time)
```

Comment:
I find this a bit messy.  You can pass a custom session to the client, why not create a custom client with the failure injection session for this test?